src/recon_buildblock/ProjMatrixByBinUsingRayTracing.cxx

```diff
@@ -668,6 +669,7 @@ calculate_proj_matrix_elems_for_one_bin(
 
       phi = proj_data_info_noarccor.get_phi(bin);
       s_in_mm = proj_data_info_noarccor.get_s(bin);
+      tantheta = proj_data_info_noarccor.get_tantheta(bin);
```

Comment:
why is it repeated here? You've already set it above. (in fact, `s_in_mm`  is also set above)

Reply:
it was set before I downcast the projection data so it would be accessible outside if statement and repeated inside the if statement to get correct value from each geometry of projection data.

Reply:
`get_tantheta` is `virtual`, so downcasting or not makes no difference: it will call the same function.

Reply:
didn't know this. What would happen if there is multiple ```get_tantheta()``` for each geometry?

Reply:
The pointer points to one particular object. Calling one of its virtual functions will call the one from its type, even if the pointer is from a base type. "run-time polymorphism" :-)